Trippy/View Controllers/Itinerary Tab/EditingItineraryViewController.m

```diff
@@ -51,19 +52,21 @@ - (void)viewDidLoad {
     self.mutableItinerary = [[Itinerary alloc] initWithDictionary:[self.baseItinerary toRouteDictionary]
                                                          prefJson:[self.baseItinerary toPrefsDictionary]
                                                         departure:self.baseItinerary.departureTime
-                                                mileageConstraint:self.baseItinerary.mileageConstraint
+                                                mileageConstraint:self.baseItinerary.mileageConstraint budgetConstraint:self.baseItinerary.budgetConstraint
                                                  sourceCollection:self.baseItinerary.sourceCollection
                                                    originLocation:self.baseItinerary.originLocation
                                                              name:self.baseItinerary.name];
     
+    self.cacheHandler = [[CacheDataHandler alloc] init];
```

Comment:
This isn't the first one of these you're alloc/initing, is it? I wonder if it should be a singleton, or else passed in?

Reply:
Not the first, but it's currently set up using delegates, so I'm not sure a singleton/passing it in would make sense?

Reply:
Yeah, with delegates that makes sense. You might want to make sure that there aren't going to be two of these things trying to interact with Parse at the same time (unless Parse handles parallel / multithreaded requests well).